### PR TITLE
Dump clif with --emit llvm-ir in release mode

### LINKED ISSUE #1020: Make --emit llvm-ir emit clif ir
This can be useful when testing.

### PR DESCRIPTION
**Pull request by osa1**
Currently it's not possible to dump llvm-ir in release build. With this
patch we allow it with `--emit llvm-ir`. In debug build we dump clif
always as before.

Fixes #1020

### CODE REVIEW

File: src/base.rs [hunk 1]
```diff
@@ -77,8 +78,9 @@ pub(crate) fn trans_fn<'clif, 'tcx, B: Backend + 'static>(
     let local_map = fx.local_map;
     let cold_blocks = fx.cold_blocks;
 
-    #[cfg(debug_assertions)]
-    crate::pretty_clif::write_clif_file(cx.tcx, "unopt", instance, &context.func, &clif_comments, None);
+    if cfg!(debug_assertions) || cx.tcx.sess.opts.output_types.contains_key(&OutputType::LlvmAssembly) {
```
Comment (bjorn3) [inline src/base.rs:81]: Maybe move this `if` into `write_clif_file`?

Comment (osa1) [reply src/base.rs:81]: Will do! :+1:

Comment (osa1) [reply src/base.rs]: Done.

File: src/pretty_clif.rs [hunk 2]
```diff
@@ -207,6 +208,10 @@ pub(crate) fn write_clif_file<'tcx>(
     mut clif_comments: &CommentWriter,
     value_ranges: Option<&cranelift_codegen::ValueLabelsRanges>,
 ) {
+    if !(cfg!(debug_assertions) || tcx.sess.opts.output_types.contains_key(&OutputType::LlvmAssembly)) {
```
Comment (bjorn3) [inline src/pretty_clif.rs:211]: I have a slight preference for

```suggestion
    if !cfg!(debug_assertions) && !tcx.sess.opts.output_types.contains_key(&OutputType::LlvmAssembly) {
```

**Comment by bjorn3** [comment]
Thanks!

**Pull request merged**